receiver/prometheusreceiver/config.go

```diff
@@ -47,6 +58,7 @@ type Config struct {
 	BufferCount             int                      `mapstructure:"buffer_count"`
 	UseStartTimeMetric      bool                     `mapstructure:"use_start_time_metric"`
 	StartTimeMetricRegex    string                   `mapstructure:"start_time_metric_regex"`
+	pdataDirect             bool
```

Comment:
There is only a global feature gate registry, so it can't necessarily be used in tests.  The feature gate registry also has a lock, so we recommend that the feature status is checked once and stored if it is needed repeatedly.  There is no mechanism for operators to change feature gate statuses during program runtime, so it should be safe to read this once and re-use it.